src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -254,6 +254,37 @@ def get_inventories(hosts: Optional[List[str]] = None,
     return inventory_hosts
 
 
+def _check_orchestrator_host_op(orch_client, hostname, add_host=True):  # pragma:no cover
+    """Check if we can adding or removing a host with orchestrator
+
+    :param orch_client: Orchestrator client
+    :param add: True for adding host operation, False for removing host
+    :raise DashboardException
+    """
+    host = orch_client.hosts.get(hostname)
+    if add_host and host:
+        raise DashboardException(
+            code='orchestrator_add_existed_host',
+            msg='{} is already in orchestrator'.format(hostname),
+            component='orchestrator')
+    if not add_host and not host:
+        raise DashboardException(
+            code='orchestrator_remove_nonexistent_host',
+            msg='Remove a non-existent host {} from orchestrator'.format(hostname),
+            component='orchestrator')
```

Comment:
Actually this is short-cutting the REST API. Don't we have a `GET /host/<host_id>` method?

Additionally, keep in mind the concept of [idempotence in HTTP](https://nordicapis.com/understanding-idempotency-and-safety-in-api-design/):
* POST ("add") is not idempotent. So strictly speaking, if you launch a POST multiple times with the same node, the API should *ADD* the node multiple times. That said, probably this is not a behaviour we really want here, so we can either ignore the situation (make POST idempotent, so adding the same node results in a 20x, as soon as the parameters are the same) or return an error (perhaps a [409](https://www.ietf.org/rfc/rfc2616.html#section-10.4.11)).
* With the [`DELETE`](https://www.ietf.org/rfc/rfc2616.html#page-56) it happens just the opposite situation. In that very case, deleting a non-added host should return a 404 (not found).

Are we sure those exceptions turn into these 409 and 404 methods or just a 500?

Reply:
I removed these changes as it is not related to this bug fix. I only intended it as a cleanup. I'll follow it up on a separate tracker: https://tracker.ceph.com/issues/51252

Reply:
> Are we sure those exceptions turn into these 409 and 404 methods or just a 500?

@epuertat @nizamial09 if http_status_code is not set in `DashboardException` then you get a 400 by default,
so we should explicitly set the appropriate status code.